packages/grafana-ui/src/components/Text/Text.tsx

```diff
@@ -0,0 +1,130 @@
+import { css, cx } from '@emotion/css';
+import React, { createElement, CSSProperties, useCallback } from 'react';
+
+import { GrafanaTheme2, ThemeTypographyVariantTypes } from '@grafana/data';
+
+import { useStyles2 } from '../../themes';
+
+interface TextProps {
+  /** Defines what HTML element is defined underneath, also maps into a default typography variant */
+  as: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6' | 'span' | 'p' | 'legend';
```

Comment:
If you do this, you should do the heading components in a seperate PR.

I'm not completely sure whether it would be okay to allow `<Text variant="h1">` to output as a span, but I _think_ it would be okay (but just strongly discouraged)

Reply:
Ill comment on some of these prop decisions since they came about from the explore task 😄 

About the `as`, this is a common pattern, in fact its one of the very few props that almost every single design system I checked implemented (see [this sheet](https://docs.google.com/spreadsheets/d/1yq4K-JjS-8eAM6XhxHiCHwKj2S9WOA4A2WTEhxHVxnc/edit#gid=0)). I think this is a more straight forward solution than having to write `<h1><Text>Heading</Text></h1>` because these are very common. The main alternative I saw is that some places have a `Text` and a `Heading` component, but then we would come to the same issue with the `Heading` implementing some sort of heading number prop and setting the variant based on that.

Reply:
I think instead we would just have `<H1 />`, `<H2 />` (notice capital H, these are our components) that come prebound with the corrent variant. `<H3>Alert rules</H3>`

We can reuse `getTextStyles` between these to make this less of a maintance burden.

Reply:
I don't have a strong preference for either, I can see how the code would be simpler with the separate components, but the docs simpler with just one component. We could ask on saga channel to see if people have strong feelings for either?